### DISCUSSION TITLE
repository_dispatch and OAuth scope - should be able to use repo:public_repo for public repositories

### DISCUSSION THREAD

**Discussion by rbellamy** [community/community#25182] [Actions]
<p>Feature request: allow “repo:public_repo” OAuth scope for Personal Access Tokens used to target repository_dispatch events for public repositories.</p>
<p>The current requirement for “repo” OAuth scope is overkill.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/rbellamy">@rbellamy</a> ,</p>
<p>Thank you for being here! I created a personal access token with only “repo:public_repo” OAuth scope, and successfully use it to create respository dispatch event to a publich repository. Could you please clarify what do you mean ‘overkill’? Thanks.</p>

**Comment by rbellamy** [comment]
<p>Well, shoot, I thought I’d tested “repo:public_repo” for triggering repository_dispatch on a public repo, and got an authorization error. Marking this topic solved.</p>
